protected/humhub/modules/search/controllers/MentioningController.php

```diff
@@ -33,37 +43,167 @@ public function getAccessRules()
+
+        return $this->asJson($results);
+    }
+
+    /**
+     * Find space members on mentioning request from RichText editor on Post form
+     *
+     * @return \yii\web\Response
+     * @throws HttpException
+     */
+    public function actionPostSpaceMembers()
+    {
+        $spaceId = (int)Yii::$app->request->get('id');
         $keyword = (string)Yii::$app->request->get('keyword');
 
-        // Add user results
-        $query = User::find()->visible()->search($keyword);
-        foreach ($query->limit(10)->all() as $container) {
-            $results[] = [
-                'guid' => $container->guid,
-                'type' => 'u',
-                'name' => $container->getDisplayName(),
-                'image' => UserImage::widget(['user' => $container, 'width' => 20]),
-                'link' => $container->getUrl()
-            ];
-        };
-
-        // Add space results
-        $query = Space::find()->visible()->search($keyword);
-        foreach ($query->limit(10)->all() as $container) {
-            $results[] = [
-                'guid' => $container->guid,
-                'type' => 's',
-                'name' => $container->getDisplayName(),
-                'image' => SpaceImage::widget(['space' => $container, 'width' => 20]),
-                'link' => $container->getUrl()
-            ];
+        $space = Space::findOne(['id' => $spaceId]);
+        if (!$space || !$space->can(CreatePost::class)) {
+            throw new HttpException(403, 'Access denied!');
+        }
+
+        // Find space members
+        $users = User::find()
+            ->leftJoin('space_membership', 'user.id = space_membership.user_id')
+            ->andWhere(['space_membership.space_id' => $spaceId])
+            ->andWhere(['space_membership.status' => Membership::STATUS_MEMBER])
+            ->visible()
+            ->search($keyword)
+            ->limit($this->module->mentioningSearchBoxResultLimit)
+            ->orderBy(['space_membership.last_visit' => SORT_DESC])
+            ->all();
+
+        $results = [];
+        foreach ($users as $user) {
+            $results[] = $this->getUserResult($user);
         }
+
+        $results = $this->appendMentioningSpaceResults($keyword, $results);
+
         return $this->asJson($results);
     }
 
+    /**
+     * Find users followed to the Content on mentioning request from RichText editor on Comment form
+     *
+     * @return \yii\web\Response
+     * @throws HttpException
+     * @throws \yii\base\Exception
+     * @throws \yii\base\InvalidConfigException
+     */
+    public function actionCommentContentFollowers()
+    {
+        $modelClass = (string)Yii::$app->request->get('model');
+        $modelId = (int)Yii::$app->request->get('id');
+        $keyword = (string)Yii::$app->request->get('keyword');
+
+        /** @var \humhub\modules\comment\Module $module */
+        $module = Yii::$app->getModule('comment');
+
+        if (!class_exists($modelClass) ||
```

Comment:
If we use `content.id` as param, it should be easier to instance the corresponding `ContentActiveRecord` (followable).

$content = humhub\modules\content\models\Content
e.g. `$post = $content->getPolymorophicRelaltion'

Reply:
Done.